### PR TITLE
Deferred: Stop inventing jQuery.when() resolution values

### PR DESCRIPTION
**Pull request by gibson042**
Fixes gh-3442

### CODE REVIEW

File: src/deferred.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ function adoptValue( value, resolve, reject ) {
 
 			// Support: Android 4.0 only
 			// Strict mode functions invoked without .call/.apply get global-object context
-			resolve.call( undefined, value );
+			resolve.apply( undefined, [ value ].slice( noValue ) );
```
Comment (mgol) [inline src/deferred.js]: This looks like "clever code", this passing booleans to slice. I guess that's done to reduce size (a regular if-else would be larger) but at least an explanatory comment would be useful.

Comment (gibson042) [inline src/deferred.js]: Good point, and correct. Updated.

File: src/deferred.js [hunk 2]
```diff
@@ -41,7 +42,7 @@ function adoptValue( value, resolve, reject ) {
 
 		// Support: Android 4.0 only
 		// Strict mode functions invoked without .call/.apply get global-object context
-		reject.call( undefined, value );
+		reject.apply( undefined, [ value ] );
```
Comment (mgol) [inline src/deferred.js]: What's the reason for this change?

Comment (gibson042) [inline src/deferred.js]: Consistency with the preceding block, which also effects size reduction (reuse of the `.apply( undefined, [ value ]…` pattern is good for compression).
